### ISSUE TITLE
Idea - enum for state?

### ISSUE THREAD

**Issue by msadowski** [tue-robotics/node_alive#3]
Hi!

Thanks for making this package publicly available! I'm wondering whether it would be useful to add one more field to the diagnostic array status that specifies the status of the node.

I can see that currently there is statusMsg.level, but by itself it doesn't allow to check whether the node is alive or if it was closed cleanly. The only other way I can think of for checking the state is to look at the statusMsg.message field.

The DiagnosticStatus (http://docs.ros.org/kinetic/api/diagnostic_msgs/html/msg/DiagnosticStatus.html) has a KeyValue array that perhaps could be reused?

**Comment by LoyVanBeek** [comment]
From what I see in the code is that the node_alive_server does A [```rosnode ping -a```](https://github.com/tue-robotics/node_alive/blob/master/src/node_alive_server#L74) which check if a node exists but does not get any information about why a node isn't running anymore.

I agree knowing whether a node existed cleanly or crashed would be very helpful but I don't think this node is capable of that in the current concept. It can only check alive/not alive.

**Comment by msadowski** [comment]
@LoyVanBeek I agree that the rosnode ping -a wouldn't give you this information however it seems that the software keeps track of nodes that has been alive and based on that it can deduct what happened to the node. Here is the function I was thinking about when creating this issue:

```
def generate_diagnostic_array(self):
        diagnosticArray = diagnostic_msgs.msg.DiagnosticArray()
        diagnosticArray.header.stamp = rospy.get_rostime()

        for tracked_node in self.tracked_nodes:

            # Create diagnostic mesasge
            statusMsg = diagnostic_msgs.msg.DiagnosticStatus()
            statusMsg.name = "node_alive%s" % tracked_node

            # Determine node status
            if tracked_node in self.alive_nodes:
                statusMsg.level = 0 # 1. Node is alive --> Status OK
                statusMsg.message = "is alive"
            else:
                if tracked_node not in self.listed_nodes:
                    if tracked_node in self.seen_nodes:
                        self.tracked_nodes.remove(tracked_node)
                        rospy.loginfo("Removed '%s' from node_alive_server" % tracked_node)
                        statusMsg.level = 0 # 2. Node has been cleanly removed
                        statusMsg.message = "cleanly removed"
                    else:
                        statusMsg.level = 2 # 3. Node did not start at all --> Status ERROR
                        statusMsg.message = "did not start"
                else:
                    statusMsg.level = 2 # 4. Node crashed --> Status ERROR
                    statusMsg.message = "is dead"

            diagnosticArray.status.append(statusMsg)

return diagnosticArray
```

It seems that we could be able to guess that the node is either:

1. Alive
2. Cleanly removed
3. Not started
4. Dead

without necessarily needing to change the logic of the software.

**Comment by reinzor** [comment]
Hi @msadowski , It makes sense to add this as a [KeyValue](http://docs.ros.org/api/diagnostic_msgs/html/msg/KeyValue.html) in the status message. Feel free to open a pull request.

**Comment by msadowski** [comment]
Closing this, as it has been resolved in #4 .
